includes/Core/User_Input/User_Input.php

```diff
@@ -0,0 +1,314 @@
+			$this->user_specific_questions->get()
+		);
+
+		// If there are no settings, return default empty values.
+		if ( empty( $settings ) ) {
+			array_walk(
+				$questions,
+				function ( &$question ) {
+					$question['values'] = array();
+				}
+			);
+
+			return $questions;
+		}
+
+		$user_id = get_current_user_id();
+
+		foreach ( $settings as &$setting ) {
+			if ( ! isset( $setting['answeredBy'] ) ) {
+				continue;
+			}
+
+			$answered_by = intval( $setting['answeredBy'] );
+			unset( $setting['answeredBy'] );
+
+			if ( ! $answered_by || $answered_by === $user_id ) {
+				continue;
+			}
+
+			$setting['author'] = array(
+				'photo' => get_avatar_url( $answered_by ),
+				'name'  => get_the_author_meta( 'user_email', $answered_by ),
+			);
+		}
+
+		// If there are un-answered questions, return default empty values for them.
+		foreach ( $questions as $question_key => $question_value ) {
+			if ( ! isset( $settings[ $question_key ] ) ) {
+				$settings[ $question_key ]           = $question_value;
+				$settings[ $question_key ]['values'] = array();
+			}
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Determines whether the current user input settings have empty values or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings The settings to check.
+	 * @return boolean|null TRUE if at least one of the settings has empty values, otherwise FALSE.
+	 */
+	public function are_settings_empty( $settings = array() ) {
+		if ( empty( $settings ) ) {
+			$settings = $this->get_answers();
+
+			if ( is_wp_error( $settings ) ) {
+				return null;
+			}
+		}
+
+		$empty_settings = array_filter(
+			$settings,
+			function( $setting ) {
+				return empty( $setting['values'] );
+			}
+		);
+
+		return 0 < count( $empty_settings );
+	}
+
+	/**
+	 * Sets user input answers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings User settings.
+	 * @return array|WP_Error User input answers.
+	 */
+	public function set_answers( $settings ) {
+		$questions     = static::$questions;
+		$site_settings = array();
+		$user_settings = array();
+
+		foreach ( $settings as $setting_key => $answers ) {
+			$setting_data           = array();
+			$setting_data['values'] = $answers;
+			$setting_data['scope']  = $questions[ $setting_key ]['scope'];
+
+			if ( 'site' === $setting_data['scope'] ) {
+				$setting_data['answeredBy']    = get_current_user_id();
+				$site_settings[ $setting_key ] = $setting_data;
+			} elseif ( 'user' === $setting_data['scope'] ) {
+				$user_settings[ $setting_key ] = $setting_data;
+			}
+		}
+
+		$this->site_specific_questions->set( $site_settings );
+		$this->user_specific_questions->set( $user_settings );
+
+		$updated_settings = $this->get_answers();
+		$is_empty         = $this->are_settings_empty( $updated_settings );
+
+		if ( ! is_null( $is_empty ) ) {
+			$this->authentication->get_user_input_state()->set( $is_empty ? User_Input_State::VALUE_MISSING : User_Input_State::VALUE_COMPLETED );
+		}
```

Comment:
We probably shouldn't be doing anything with `User_Input_State` since it is going to be removed as part of this epic. This also seems to be the only reason for this class as a dependency. If we need to, let's wait on #5900 before resuming work on this.

Reply:
Understood, thank you, @aaemnnosttv! The only reason we're doing this because without it, the `User_Input_State` isn't set to a `completed` state, hence the user is stuck in the user-input screen without being able to navigate to the dashboard.

Also, #5900 is again blocked by #6155 (the issue this PR addresses). How do you think we should proceed regarding this? Thanks!

Reply:
Hmm, as a temporary workaround I think we can add an action here, e.g. `googlesitekit_user_input_set` (passing along `$is_empty` as a param) and then hook into that from `User_Input_State::register` (we'd need to override it to do this). Then when that class is removed we can remove the action here as well. If we do that then we should make a note on that issue to make sure it gets cleaned up in the process.

Reply:
Updated to add the hook. Also added a note in #5900 [here](https://github.com/google/site-kit-wp/issues/5900). Thank you!